### PR TITLE
Allow `schema` to be omitted when using default root op names

### LINKED ISSUE #304: Populating entry points without explicity schema{}
From https://facebook.github.io/graphql/June2018/#sec-Root-Operation-Types : 

> While any type can be the root operation type for a GraphQL operation, the type system definition language can omit the schema definition when the query, mutation, and subscription root types are named Query, Mutation, and Subscription respectively.
> 
> Likewise, when representing a GraphQL schema using the type system definition language, a schema definition should be omitted if it only uses the default root operation type names.
> 
> This example describes a valid complete GraphQL schema, despite not explicitly including a schema definition. The Query type is presumed to be the query root operation type of the schema.

If I understand this correctly, having a type named `type Query` with no `schema {}` should be the same as doing `type Query ... schema { query: Query }` but currently this does not work.

### PR DESCRIPTION
**Pull request by dackroyd**
Fixes #304

### CODE REVIEW

File: graphql_test.go [hunk 1]
```diff
@@ -2512,15 +2742,19 @@ func (r *inputResolver) Nullable(args struct{ Value *int32 }) *int32 {
 	return args.Value
 }
 
-func (r *inputResolver) List(args struct{ Value []*struct{ V int32 } }) []int32 {
+func (r *inputResolver) List(args struct {
```
Comment (dackroyd) [inline graphql_test.go]: Crept in, will revert these unrelated changes

File: internal/exec/resolvable/resolvable.go [hunk 2]
```diff
@@ -362,7 +362,8 @@ func (b *execBuilder) makeFieldExec(typeName string, f *schema.Field, m reflect.
 	var out reflect.Type
 	if methodIndex != -1 {
 		out = m.Type.Out(0)
-		if typeName == "Subscription" && out.Kind() == reflect.Chan {
+		sub, ok := b.schema.EntryPoints["subscription"]
```
Comment (dackroyd) [inline internal/exec/resolvable/resolvable.go]: This is an existing bug which prevents the `subscription` root op having a type named anything other than `Subscription`. Had to fix it here to verify correct behaviour for custom type names, but can split that work if desired

**Comment by dackroyd** [comment]
Looking through open PRs on the repo, I notice that this has been raised previously under #126 back in 2017, but never merged. The codebase has drifted quite a lot since then, so it has picked up a few conflicts

File: graphql.go [hunk 3]
```diff
@@ -223,6 +231,39 @@ func (s *Schema) exec(ctx context.Context, queryString string, operationName str
...
+
+func validateRootOp(s *schema.Schema, name string, mandatory bool) error {
+	t, ok := s.EntryPoints[name]
+	if !ok {
+		if mandatory {
+			return fmt.Errorf("root operation %q must be defined", name)
+		}
+		return nil
+	}
+	if t.Kind() != "OBJECT" {
```
Comment (dackroyd) [inline graphql.go]:
```
if _, ok := t.(*schema.Object); !ok {
```

^ could be a better way of checking this?

Comment (pavelnikolov) [reply graphql.go]: They are equivalent, IMHO
